fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalOlapScanOperator.java

```diff
@@ -44,6 +44,9 @@ public final class LogicalOlapScanOperator extends LogicalScanOperator {
 
     private List<ScalarOperator> prunedPartitionPredicates;
 
+    // record if this scan is derived from SplitScanORToUnionRule
+    private boolean fromSplitOR;
```

Comment:
why need this mark?

Reply:
It's for the pattern check fast return. To aviod a the new scan child enter the rule again.